include/mbgl/storage/response.hpp

```diff
@@ -45,15 +46,24 @@ class Response::Error {
         NotFound = 2,
         Server = 3,
         Connection = 4,
+        RateLimit = 5,
         Other = 6,
     } reason = Reason::Other;
 
     // An error message from the request handler, e.g. a server message or a system message
     // informing the user about the reason for the failure.
     std::string message;
+    
+    optional<Timestamp> retryAfter;
 
 public:
-    Error(Reason, std::string = "");
+    //Either; a HttpDate, seconds from now or a Timestamp
+    using RetryAfter = mbgl::variant<std::string, std::chrono::seconds, Timestamp>;
+    
+    Error(Reason, std::string = "", optional<RetryAfter> = {});
+    
+private:
+    optional<Timestamp> parseRetryAfter(const optional<RetryAfter> retryAfter_) const;
```

Comment:
Can you move this to `http_header.hpp` and make the signature:

```
optional<Timestamp> parseRetryHeaders(const optional<std::string>& retryAfter,
                                      const optional<std::string>& xRateLimitReset);
```

Advantages:
* No need for `RetryAfter` type
* Factor conditional and integer conversion logic from individual SDK implementations